nengo/tests/test_processes.py

```diff
@@ -538,3 +539,310 @@ def func(t):
         with pytest.warns(UserWarning):
             process = Piecewise({0.05: 0, 0.1: func}, interpolation="linear")
         assert process.interpolation == "zero"
+
+
+class TestLinearSystem:
+    def run_sim(
+        self,
+        Simulator,
+        sys,
+        analog=True,
+        x0=0,
+        dt=0.001,
+        simtime=0.3,
+        plt=None,
+        f_in=None,
+    ):
+        process = (
+            sys
+            if isinstance(sys, LinearSystem)
+            else LinearSystem(sys, analog=analog, x0=x0)
+        )
+
+        with nengo.Network() as model:
+            y = nengo.Node(process)
+            yp = nengo.Probe(y)
+
+            if f_in is not None:
+                u = nengo.Node(f_in)
+                nengo.Connection(u, y, synapse=None)
+
+        with Simulator(model, dt=dt, progress_bar=False) as sim:
+            sim.run(simtime)
+
+        if plt:
+            plt.plot(sim.trange(), sim.data[yp])
+
+        return sim.trange(), sim.data[yp]
+
+    def test_passthrough(self, Simulator, plt, allclose):
+        f_in = lambda t: np.array([0.3, 0.7]) * np.sin(5 * t)
+        dt = 0.001
+        t, y = self.run_sim(
+            Simulator, (None, None, None, [[0, 2], [3, 0]]), dt=dt, plt=plt, f_in=f_in
+        )
+        u = f_in(t[:, None])
+        assert allclose(y[:, 0], 2 * u[:, 1], atol=1e-4)
+        assert allclose(y[:, 1], 3 * u[:, 0], atol=1e-4)
+
+    def test_autonomous_oscillator(self, Simulator, plt, allclose):
+        omega = 6 * np.pi
+        A = [[0, -omega], [omega, 0]]
+        C = np.eye(2)
+        x0 = [1, 0]
+
+        dt = 0.001
+        t, y = self.run_sim(
+            Simulator, (A, None, C, None), plt=plt, x0=x0, dt=dt, simtime=1.0
+        )
+        t -= dt
+        assert allclose(y[:, 0], np.cos(omega * t), atol=1e-4)
+        assert allclose(y[:, 1], np.sin(omega * t), atol=1e-4)
+
+    @pytest.mark.parametrize("dims", [1, 3])
+    def test_integrator(self, dims, Simulator, rng, plt, allclose):
+        dt = 0.001
+        A = None
+        B = np.eye(dims)
+        C = np.eye(dims)
+
+        freqs = rng.uniform(5 * np.pi, 7 * np.pi, size=dims)
+        amps = rng.uniform(0.5, 1.5, size=dims)
+        f_in = lambda t: amps * np.sin(freqs * t)
+
+        t, y = self.run_sim(Simulator, (A, B, C, None), dt=dt, plt=plt, f_in=f_in)
+        t -= dt
+        u = f_in(t[:, None])
+        assert allclose(y, np.cumsum(u, axis=0) * dt, atol=1e-4)
+
+    def test_discretize(self, Simulator, plt):
+        dt = 0.003
+        sys = LinearSystem(([1], [0.02, 1]), method="euler")
+        f_in = lambda t: np.sin(20 * t)
+        t, y0 = self.run_sim(Simulator, sys, dt=dt, plt=plt, f_in=f_in)
+        t, y1 = self.run_sim(Simulator, sys.discretize(dt), dt=dt, plt=plt, f_in=f_in)
+        t, y2 = self.run_sim(
+            Simulator, sys.discrete_ss(dt), analog=False, dt=dt, plt=plt, f_in=f_in
+        )
+        assert np.allclose(y1, y0)
```

Comment:
Are these and other `np.allclose` calls intentionally not using the `allclose` fixture?

Reply:
No, just an oversight I think. I'd definitely change the ones in this test to `allclose`, because it uses `Simulator`. The others don't matter so much because they don't use `Simulator`, so they aren't for backends. But there's no harm in using `allclose`, I don't think.